sdk/iot/Azure.Iot.Hub.Service/src/ModulesClient.cs

```diff
@@ -160,6 +161,202 @@ public virtual Response<IReadOnlyList<ModuleIdentity>> GetIdentities(string devi
+        /// </summary>
+        /// <param name="moduleIdentities">The module identities to create.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The result of the bulk operation and the http response <see cref="Response{T}"/>.</returns>
+        public virtual Task<Response<BulkRegistryOperationResponse>> CreateIdentitiesAsync(IEnumerable<ModuleIdentity> moduleIdentities, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<ExportImportDevice> registryOperations = moduleIdentities
+                .Select(x => new ExportImportDevice()
+                {
+                    Id = x.DeviceId,
+                    ModuleId = x.ModuleId,
+                    Authentication = x.Authentication,
+                    ImportMode = ExportImportDeviceImportMode.Create
+                });
+
+            return _registryManagerClient.BulkDeviceCrudAsync(registryOperations, cancellationToken);
+        }
+
+        /// <summary>
+        /// Create multiple modules. A maximum of 100 creations can be done per call, and each module identity must be unique.
+        /// All devices that these modules will belong to must already exist. Multiple modules can be created at a time on a single device.
+        /// For larger scale operations, consider using <see href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities">IoT Hub jobs</see>.
+        /// </summary>
+        /// <param name="moduleIdentities">The module identities to create.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The result of the bulk operation and the http response <see cref="Response{T}"/>.</returns>
+        public virtual Response<BulkRegistryOperationResponse> CreateIdentities(IEnumerable<ModuleIdentity> moduleIdentities, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<ExportImportDevice> registryOperations = moduleIdentities
+                .Select(x => new ExportImportDevice()
+                {
+                    Id = x.DeviceId,
+                    ModuleId = x.ModuleId,
+                    Authentication = x.Authentication,
+                    ImportMode = ExportImportDeviceImportMode.Create
+                });
+
+            return _registryManagerClient.BulkDeviceCrud(registryOperations, cancellationToken);
+        }
+
+        /// <summary>
+        /// Update multiple modules. A maximum of 100 updates can be done per call, and each operation must be done on a different identity. For larger scale operations, consider using <see href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities">IoT Hub jobs</see>..
```

Comment:
I assume you mean " each operation must be done on a different <module> identity"? We should maybe call that out?

It looks like we can only update the auth mode, should we call that out explicitly as well?

Reply:
I'm confused, I wrote "each operation must be done on a different identity" in there. What do you mean?

Reply:
> It looks like we can only update the auth mode, should we call that out explicitly as well?

Since that is a service side decision, I'd rather not

Reply:
> I'm confused, I wrote "each operation must be done on a different identity" in there. What do you mean?

I wrote ` each operation must be done on a different <module> identity`, and since I put "modules" within `<>`, the .md formatting hid that.

Reply:
> > It looks like we can only update the auth mode, should we call that out explicitly as well?
> 
> Since that is a service side decision, I'd rather not

The way we expose the API, we would still have to update our APIs if service ever changes their behavior. 
My only concern is, we call it "update identities", but the only thing updateable is the auth mechanism, and a user wouldn't know that until they look into the implementation.